src/html5_parser/node_data/comment_data.rs

```diff
@@ -0,0 +1,33 @@
+#[derive(Debug, PartialEq, Clone)]
+pub struct CommentData {
+    value: String,
+}
+
+impl Default for CommentData {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl CommentData {
+    pub fn new() -> Self {
+        CommentData {
+            value: "".to_string(),
+        }
+    }
+
+    pub fn new_with_value(value: &str) -> Self {
+        let mut comment_data = CommentData::new();
+        comment_data.set_value(value);
+
+        comment_data
+    }
+
+    pub fn get_value(&self) -> String {
+        self.value.clone()
+    }
```

Comment:
```suggestion
    pub fn value(&self) -> &str {
        &self.value
    }
```

Rust knows the difference between when a field on a struct is being accessed and when a function with the same name is being called. I have not seen readers with the `get_` prefix very often.

In this case, it would also be nice to let the caller opt into cloning the string instead of preemptively doing it.